### PR TITLE
feat: add new crate op-beacon-core

### PR DESCRIPTION
**Pull request by rupam-04**
Addresses #7648 

Would want some insights on the current work if I'm going in the right direction or not

### CODE REVIEW

**Comment by rupam-04** [comment]
Also I'd like to know what is causing the following error:

```
error: failed to load manifest for workspace member `/home/runner/work/reth/reth/crates/op-beacon-core`

Caused by:
  failed to parse manifest at `/home/runner/work/reth/reth/crates/op-beacon-core/Cargo.toml`

Caused by:
  error inheriting `reth-consensus-beacon-core` from workspace root manifest's `workspace.dependencies.reth-consensus-beacon-core`

Caused by:
  `dependency.reth-consensus-beacon-core` was not found in `workspace.dependencies`
```

File: Cargo.toml [hunk 1]
```diff
@@ -84,7 +84,7 @@ members = [
     "examples/custom-inspector/",
     "examples/exex/minimal/",
     "examples/exex/op-bridge/",
-    "testing/ef-tests/",
+    "testing/ef-tests/", "crates/op-beacon-core",
```
Comment (mattsse) [inline Cargo.toml:87]: these needs to be moved up to where other crates are defined

File: Cargo.toml
Comment (mattsse) [inline Cargo.toml:1]: the crate also needs to be added here this should fix the error

https://github.com/paradigmxyz/reth/blob/86980836b8cc7e36ecf47400485f544cd2ffe2eb/Cargo.toml#L201-L201

File: crates/op-beacon-core/src/lib.rs
Comment (mattsse) [inline crates/op-beacon-core/src/lib.rs:1]: this needs the regular lib.rs header, see other crates for ref

**Review by mattsse — CHANGES_REQUESTED** [review]
this is a great first step,

I also wanto to move this to the `crates/optimism` folder

File: Cargo.toml
Comment (rupam-04) [reply Cargo.toml:1]: Having some issues with this. I added both
```
reth-op-beacon-core = { path = "crates/optimism/op-beacon-core" }
reth-consensus-beacon-core = { path = "crates/consensus/beacon-core" }
``` to ```reth/Cargo.toml``` but there still seems to be some issues.

Comment (rupam-04) [reply Cargo.toml:1]: This is the error I get this time:

```
error: no matching package named `reth-consensus-beacon-core` found
location searched: /home/runner/work/reth/reth/crates/consensus/beacon-core
required by package `op-beacon-core v0.2.0-beta.6 (/home/runner/work/reth/reth/crates/optimism/op-beacon-core)`
```

Comment (onbjerg) [reply Cargo.toml:1]: it's called reth-beacon-consensus-core, not reth-consensus-beacon-core

https://github.com/paradigmxyz/reth/blob/4f81f3acc9fdc56b18444c6754f852b7060c57ee/crates/consensus/beacon-core/Cargo.toml#L2

File: crates/optimism/op-beacon-core/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+version.workspace = true
+edition.workspace = true
+rust-version.workspace = true
+license.workspace = true
+homepage.workspace = true
+repository.workspace = true
+exclude.workspace = true
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
```
Comment (onbjerg) [inline crates/optimism/op-beacon-core/Cargo.toml:12]:
```suggestion
```

nit

File: crates/optimism/op-beacon-core/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
+//! Beacon consensus implementation
```
Comment (onbjerg) [inline crates/optimism/op-beacon-core/src/lib.rs:1]:
```suggestion
//! Optimism beacon consensus implementation
```

**Review by onbjerg — COMMENTED** [review]
also please run `cargo +nightly fmt --all`

**Comment by rupam-04** [comment]
Few conflicts due to the changed file location of ```crates/interfaces/src/consensus.rs``` probably due to #7813 @mattsse

**Comment by rupam-04** [comment]
> Hey @rupam-04 it seems you accidentally force pushed main onto the branch, so the PR auto closed. Feel free to re-open a PR for this

Yeah it had some conflicts, I reopened a PR

**Pull request closed without merging**